music21/musicxml/m21ToXml.py

```diff
@@ -3425,13 +3425,10 @@ def objectAttachedSpannersToNotations(self, obj, noteIndexInChord=0, objectSpann
                 continue
             if su.type == 'non-arpeggio':
                 # <non-arpeggiate> goes only on top and bottom note in chord
-                if not foundTopAndBottom:
-                    topNoteIndex, bottomNoteIndex = self.topAndBottomIndices(obj)
-                    foundTopAndBottom = True
-                if bottomNoteIndex == noteIndexInChord:
+                if noteIndexInChord == 0:
                     mxArpeggio = Element('non-arpeggiate')
                     mxArpeggio.set('type', 'bottom')
-                elif topNoteIndex == noteIndexInChord:
+                elif noteIndexInChord == len(obj.notes):
```

Comment:
this will never happen, off by one error.  `len(obj.notes) - 1`  can we have a test to make sure these are generated correctly.

Reply:
Yeah, my multiStaffArpeggio test doesn't have any non-arpeggiates in it.  I'll fix that (and then this bug).

Reply:
it's in my newest PR